src/components/Checkbox/view/alpha/style/Checkbox.style.ts

```diff
@@ -0,0 +1,188 @@
+import styled, { css } from 'styled-components';
+import { Size } from '../../../../../types';
+import { StyledIcon } from '../../../../Icon';
+import { getDefaultTypographyStyles } from '../../../../Typography/style';
+
+type StyledCheckboxProps = {
+    isChecked?: boolean;
+    isPartiallyChecked?: boolean;
+    isDisabled?: boolean;
+    isTextDotted?: boolean;
+    name?: string;
+    position?: 'right' | 'left';
+    size?: Size;
+};
+
+export const HiddenCheckbox = styled.input.attrs({
+    type: 'checkbox',
+})`
+    opacity: 0;
+    width: 0;
+    height: 0;
+    margin: 0;
+    position: absolute;
+    z-index: -1;
+`;
+
+export const StyledCheckboxInputShown = styled('div').withConfig({
+    shouldForwardProp: (prop) => !['isDisabled'].includes(prop),
+})<StyledCheckboxProps>`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: ${(props) => parseInt(props.theme.padding.xs, 0) - 1}px;
+    background: ${(props) => props.theme.color.background.alpha};
+    border-radius: ${(props) => parseInt(props.theme.radius, 0) - 1}px;
+    border: 1px solid ${(props) => props.theme.color.border};
+
+    ${(props) =>
+        props.isDisabled &&
+        css`
+            background-color: ${props.theme.color.default};
+            cursor: not-allowed;
+        `}
+`;
+
+export const StyledCheckboxLabel = styled.label.withConfig({
+    shouldForwardProp: (prop) => !['isDisabled', 'size'].includes(prop),
+})<StyledCheckboxProps>`
+    ${(props) => getDefaultTypographyStyles(props)};
+    display: inline-flex;
+    align-items: center;
+    height: ${(props) => props.theme.height.sm};
+    cursor: pointer;
+
+    ${(props) =>
+        !props.isDisabled &&
+        css`
+            :hover {
+                ${StyledCheckboxInputShown} {
+                    border: 1px solid ${props.theme.color.primary.alpha};
+                }
+            }
+        `};
+
+    ${(props) =>
+        props.isDisabled &&
+        css`
+            cursor: not-allowed;
+        `};
+
+    ${(props) =>
+        props.size === 'small' &&
+        css`
+            height: ${props.theme.height.sm};
+        `}
+
+    ${(props) =>
+        props.size === 'medium' &&
+        css`
+            height: ${props.theme.height.md};
+        `}
+    
+    ${(props) =>
+        props.size === 'large' &&
+        css`
+            height: ${props.theme.height.lg};
+        `}
+`;
+
+export const StyledCheckboxMark = styled.div.withConfig({
+    shouldForwardProp: (prop) =>
+        !['isChecked', 'isPartiallyChecked', 'isDisabled'].includes(prop),
+})<StyledCheckboxProps>`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: ${(props) => props.theme.width.xs};
+    min-width: ${(props) => props.theme.width.xs};
+    height: ${(props) => props.theme.width.xs};
+    min-height: ${(props) => props.theme.width.xs};
```

Comment:
Tipuji, že `min-height` a `min-width` hodnoty jsi vzala z předchozího checkboxu. To si myslím, že předtím bylo psané kvůli vlastnosti flexboxu, kdy v jistých případech deforumuje velikost elementu. To jde eliminovat i zapsáním `flex-shrink: 0`.